### ISSUE TITLE
Typesense does not return data for query with spaces

### ISSUE THREAD

**Issue by entrptaher** [typesense/typesense#633]
I am testing it on v0.23, kindly let me know if this is behaving correctly. I have **sol punks**

![image](https://user-images.githubusercontent.com/8284972/175784583-5ede96eb-b888-496f-a60d-d6b324bba055.png)

But what I also want to include in the search result is this one without space.

![image](https://user-images.githubusercontent.com/8284972/175784596-14453908-47a7-40cd-a909-1a1b980e96af.png)

If I try any other combination like this, it works, 

![image](https://user-images.githubusercontent.com/8284972/175784643-429de731-c8bc-4468-8e4c-730d79b7caf1.png)

Basically everything works except **sol punks** with a space.

When I removed those two items, it started to return data for this.

![image](https://user-images.githubusercontent.com/8284972/175785152-c632f95b-de35-4f5a-bf2b-4eca03f161eb.png)

**Comment by entrptaher** [comment]
What I want is to return data like this. This is how postgresql is returning the data using full text search. Typesense fails to do this.

![image](https://user-images.githubusercontent.com/8284972/175785201-47730a71-a308-487b-907b-b6d677a56854.png)

**Comment by kishorenc** [comment]
@entrptaher The splitting / joining of tokens only happens when there are no results for the original query. This is to prevent false positives.

**Comment by entrptaher** [comment]
I totally understand. I believe other full text search returns data for space because you can see solpunks actually includes sol and punks both words but is not returned by typesense. What can be done to return all three results?

**Comment by entrptaher** [comment]
I have tested this with meilisearch and it returns same as the postgresql full text search as you can see.

![image](https://user-images.githubusercontent.com/8284972/175804215-cae3405e-f927-414a-802f-da4a332b5fdd.png)

![image](https://user-images.githubusercontent.com/8284972/175804239-bcf76af0-3040-45de-9250-629dc1e14849.png)

This is the normal behavior returned by algolia, meilisearch, postgresql etc.

**Comment by kishorenc** [comment]
Ok, I've modified the `split_join_tokens` flag to take an `always` mode now which will give the behavior you want. 

i.e. send `?split_join_tokens=always` for splitting/joining to happen even if there are results that match query tokens directly. This is available in `typesense/typesense:0.24.0.rc10` Docker image.

**Comment by entrptaher** [comment]
Thank you @kishorenc for your hard work! Looks like it returns the three results with the said version. Let me experiment with it and see how it goes. 

![image](https://user-images.githubusercontent.com/8284972/175856557-4903523c-4265-42de-abfa-c037be7aa5ad.png)

**Comment by geisterfurz007** [comment]
@kishorenc For clarification: Was the off/fallback/always behaviour introduced in 0.24.0? It's documented for 0.23.1 as well (but not for 0.23.0).

**Comment by kishorenc** [comment]
@geisterfurz007 It's available in 0.24 RC as well as on 0.23.1 but not on 0.23.0.
